src/main/java/frc/robot/subsystems/AlgaeCollector.java

```diff
@@ -0,0 +1,169 @@
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.simulation.LinearSystemSim;
+import edu.wpi.first.wpilibj.simulation.SingleJointedArmSim;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.thunder.hardware.ThunderBird;
+import frc.thunder.shuffleboard.LightningShuffleboard;
+import frc.robot.Robot;
+import frc.robot.Constants.AlgaeCollectorConstants;
+import frc.robot.Constants.RobotMap;
+
+public class AlgaeCollector extends SubsystemBase {
+
+    private ThunderBird rollerMotor;
+    private ThunderBird pivotMotor;
+
+    private PositionVoltage pivotPID = new PositionVoltage(0);
+    private double targetAngle = 0;
+
+    private TalonFXSimState rollerMotorSim;
+    private TalonFXSimState pivotMotorSim;
+
+    private SingleJointedArmSim pivotSim;
+    @SuppressWarnings("rawtypes")
+    private LinearSystemSim rollerSim;
+
+    private DCMotor pivotGearbox;
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public AlgaeCollector(ThunderBird rollerMotor, ThunderBird pivotMotor) {
+        this.rollerMotor = rollerMotor;
+        this.pivotMotor = pivotMotor;
+
+        TalonFXConfiguration pivotMotorConfig = pivotMotor.getConfig();
+
+        if(Robot.isSimulation()){
+            // simulate motors
+            rollerMotorSim = new TalonFXSimState(rollerMotor);
+            pivotMotorSim = new TalonFXSimState(pivotMotor);
+
+            pivotGearbox = DCMotor.getKrakenX60(1);
+
+            // create physics sims
+            pivotSim = new SingleJointedArmSim(pivotGearbox, AlgaeCollectorConstants.PIVOT_GEAR_RATIO, 
+                AlgaeCollectorConstants.PIVOT_MOI, AlgaeCollectorConstants.PIVOT_LENGTH, Units.degreesToRadians(AlgaeCollectorConstants.PIVOT_MIN_ANGLE), 
+                Units.degreesToRadians(AlgaeCollectorConstants.PIVOT_MAX_ANGLE), true, AlgaeCollectorConstants.PIVOT_START_ANGLE, 0, 1);
+
+            rollerSim = new LinearSystemSim(LinearSystemId.identifyVelocitySystem(AlgaeCollectorConstants.ROLLER_KV, 
+                AlgaeCollectorConstants.ROLLER_KA));
+        }
+
+        pivotMotorConfig.Slot0.kP = AlgaeCollectorConstants.PIVOT_KP;
+        pivotMotorConfig.Slot0.kI = AlgaeCollectorConstants.PIVOT_KI;
+        pivotMotorConfig.Slot0.kD = AlgaeCollectorConstants.PIVOT_KD;
+
+        pivotMotor.applyConfig(pivotMotorConfig);
+    }
+
+    @Override
+    public void periodic() {
+        if (!DriverStation.isFMSAttached()){
+            LightningShuffleboard.setDouble("Algae Collector", "Pivot Angle", getPivotAngle());
+            LightningShuffleboard.setDouble("Algae Collector", "Roller Velocity", getRollerVelocity());
+            LightningShuffleboard.setDouble("Algae Collector", "Pivot Target", getTargetAngle());
+            LightningShuffleboard.setDouble("Algae Collector", "Pivot Motor current", pivotMotorSim.getMotorVoltage());
+            LightningShuffleboard.setDouble("Algae Collector", "Pivot raw roter pos0ition", pivotSim.getAngleRads());
+        }
```

Comment:
remove the conditional for now, we will go back and figure out what is most necessary for comp values later